From 011b9905c09017fd90ef6c608228b9bb45e789bd Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Mon, 12 Jun 2023 14:47:45 +0800
Subject: [PATCH] u-boot-32: Enable support for the legacy image format

When run script boot.scr, there is below error:
=> source
    ## Executing script at 80000000
Wrong image format for "source" command
Because the config CONFIG_LEGACY_IMAGE_FORMAT is disabled
by default if CONFIG_FIT_SIGNATURE is enabled. So, enable legacy
 image format explicitly to support run boot script.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 configs/s32g274ardb2_defconfig | 1 +
 configs/s32g2xxaevb_defconfig  | 1 +
 configs/s32g399ardb3_defconfig | 1 +
 configs/s32g3xxaevb_defconfig  | 1 +
 4 files changed, 4 insertions(+)

diff --git a/configs/s32g274ardb2_defconfig b/configs/s32g274ardb2_defconfig
index 22614ca756..7731a3ea93 100644
--- a/configs/s32g274ardb2_defconfig
+++ b/configs/s32g274ardb2_defconfig
@@ -7,3 +7,4 @@ CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_NXP_HSE_SUPPORT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
diff --git a/configs/s32g2xxaevb_defconfig b/configs/s32g2xxaevb_defconfig
index 232dbf4efd..a205d0bb62 100644
--- a/configs/s32g2xxaevb_defconfig
+++ b/configs/s32g2xxaevb_defconfig
@@ -9,3 +9,4 @@ CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_NXP_HSE_SUPPORT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
diff --git a/configs/s32g399ardb3_defconfig b/configs/s32g399ardb3_defconfig
index 9cedbb0244..dfc4482dfd 100644
--- a/configs/s32g399ardb3_defconfig
+++ b/configs/s32g399ardb3_defconfig
@@ -7,3 +7,4 @@ CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_NXP_HSE_SUPPORT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
diff --git a/configs/s32g3xxaevb_defconfig b/configs/s32g3xxaevb_defconfig
index 4a1f095c6b..09a8391c62 100644
--- a/configs/s32g3xxaevb_defconfig
+++ b/configs/s32g3xxaevb_defconfig
@@ -9,3 +9,4 @@ CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_NXP_HSE_SUPPORT=y
 CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
-- 
2.40.1

